doc/src/modules/polys/domainsintro.rst

```diff
@@ -925,7 +925,7 @@ two versions of :ref:`K[x]` using :py:meth:`~.Domain.poly_ring` and
   >>> p1
   x**2 + 1
   >>> p2
-  x**2 + 1
+  DMP([1, 0, 1], ZZ, None)
```

Comment:
Since `DMP` no longer has a `.ring` attribute it does not have any generators and cannot print as e.g. `x**2 + 1`.

Any code that just prints a raw `DMF` will not show the raw representation. Previously DMF would sometimes have a ring and sometimes not making it a  bit unreliable for printing and many other things.